ROS 2 adc node (#135)
husarion/panther_ros: panther_battery/README.md

## ROS Nodes

### adc_node

Publishes battery state read from ADC unit. Available from Panther version 1.2.

#### Publishes

> - `/panther/battery` [*sensor_msgs/BatteryState*]: average values of both batteries if panther has two batteries. In case of single battery values only for the single one.

Reviewer: ```suggestion
- `/panther/battery` [*sensor_msgs/BatteryState*]: mean values of both batteries if panther has two batteries. Otherwise state of the single battery will be published.
```

Author: Done

- `/panther/battery_1` [*sensor_msgs/BatteryState*]: first battery state. Published if second battery detected.
- `/panther/battery_2` [*sensor_msgs/BatteryState*]: second battery state. Published if second battery detected.

#### Subscribes

- `/panther/driver/motor_controllers_state` [*panther_msgs/DriverState*]: current motor controllers' state and error flags.

#### Parameters

- `~adc0_device` [*string*, default: **/sys/bus/iio/devices/iio:device0**]: ADC nr 0 device IIO path.
- `~adc1_device` [*string*, default: **/sys/bus/iio/devices/iio:device0**]: ADC nr 0 device IIO path.

> - `~battery_charge_window_len` [*int*, default: **10**]: moving average window length used to smooth out battery charge readings.

Reviewer: ```suggestion
- `~battery_charge_window_len` [*int*, default: **10**]: window length of a moving average, used to smooth out battery charge readings.
```

> - `~battery_current_window_len` [*int*, default: **10**]: moving average window length used to smooth out battery current readings.

Reviewer: ```suggestion
- `~battery_current_window_len` [*int*, default: **10**]: window length of a moving average, used to smooth out battery current readings.
```

> - `~battery_temp_window_len` [*int*, default: **10**]: moving average window length used to smooth out battery temperature readings.

Reviewer: ```suggestion
- `~battery_temp_window_len` [*int*, default: **10**]: window length of a moving average, used to smooth out battery temperature readings.
```

> - `~battery_timeout` [*float*, default: **1.0**]: specifies the timeout in seconds. If node fails to read ADC battery data exceeding this duration, the node will publish an unknown battery state.

Reviewer: As mentioned later. Not sure it this is necessary

> - `~battery_voltage_window_len` [*int*, default: **10**]: moving average window length used to smooth out battery voltage readings.

Reviewer: ```suggestion
- `~battery_voltage_window_len` [*int*, default: **10**]: window length of a moving average, used to smooth out battery voltage readings.
```

> - `~high_bat_temp` [*float*, default: **55.0**]: The temperature of the battery at which the battery health state is incorrect.

Reviewer: ```suggestion
- `~high_bat_temp` [*float*, default: **55.0**]: The temperature of the battery at which is ic considered to overheat.
```

Author: Done

#### Parameters

> - `~battery_current_window_len` [*int*, default: **10**]: moving average window length used to smooth out battery current readings.

Reviewer: ```suggestion
- `~battery_current_window_len` [*int*, default: **10**]: window length of a moving average, used to smooth out battery current readings.
```

- `~battery_timeout` [*float*, default: **1.0**]: specifies the timeout in seconds. If no new battery messages are received within this duration, the node will publish an unknown battery state.

> - `~batery_voltage_window_len` [*int*, default: **10**]: moving average window length used to smooth out battery voltage readings.

Reviewer: ```suggestion
- `~batery_voltage_window_len` [*int*, default: **10**]: window length of a moving average, used to smooth out battery voltage readings.
```
